dev-support/test-patch.d/scaladoc.sh

```diff
@@ -0,0 +1,139 @@
+#!/usr/bin/env bash
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+add_plugin scaladoc
+
+
+function scaladoc_filefilter
+{
+  local filename=$1
+
+  if [[ ${filename} =~ \.scala$ ]]; then
+   yetus_debug "tests/scaladoc: ${filename}"
+   add_test scaladoc
+  fi
+}
+
+## @description  Confirm Javadoc pre-patch
+## @audience     private
+## @stability    stable
+## @replaceable  no
+## @return       0 on success
+## @return       1 on failure
+function scaladoc_preapply
+{
+  local result=0
+
+  verify_needed_test scaladoc
+  if [[ $? == 0 ]]; then
+     return 0
+  fi
+
+  big_console_header "Pre-patch ${PATCH_BRANCH} Javadoc verification"
+
+  personality_modules branch scaladoc
+  ${BUILDTOOL}_modules_worker branch scaladoc
+
+  ((result=result + $?))
+  modules_messages branch scaladoc true
+
+
+  if [[ ${result} -gt 0 ]]; then
+    return 1
+  fi
+  return 0
+}
+
+## @description  Count and compare the number of JavaDoc warnings pre- and post- patch
+## @audience     private
+## @stability    evolving
+## @replaceable  no
+## @return       0 on success
+## @return       1 on failure
+function scaladoc_postinstall
+{
+  local i
+  local result=0
+  local fn
+  declare -i numbranch=0
+  declare -i numpatch=0
+
+  verify_needed_test scaladoc
+    if [[ $? == 0 ]]; then
+    return 0
+  fi
+
+  big_console_header "Determining number of patched scaladoc warnings"
+
```

Comment:
shellcheck: 3: note: Double quote to prevent globbing and word splitting. [SC2086]